File: js/one_edit_distance.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
 var isOneEditDistance = function(s, t) {
    let n = s.length, m = t.length;

    if(s == t){
        return false;
    }

    if(n == m){
        for(let i = 0; i < n; i++){
            let newStr = s.substring(0, i) + t.charAt(i) + s.substring(i + 1, s.length)

            if(newStr == t){
                return true;
            }
        }
        return false;
    } else if (n == m - 1){
        for(let i = 0; i < m; i++){
            let newStr = s.substring(0, i) + t[i] + s.substring(i , s.length);

            if(newStr == t){
                return true;
            }
        }

        return false;
    } else if (n - 1 == m){
        for(let i = 0; i < n; i++){
            let newStr = s.substring(0, i) + s.substring(i + 1, s.length);

            if(newStr == t){
                return true;
            }
        }

        return false;
    } else {
        return false;
    }
};

console.log('isOneEditDistance: ', isOneEditDistance('', ''))

File: cpp/array/one_edit_distance.cpp
#include<iostream>
#include<vector>

using namespace std;
    
class Solution {
public:
    bool isOneEditDistance(string s, string t) {
        int n = s.size(), m = t.size();

        if(s == t){
            return false;
        }

        if(n == m){
            for(int i = 0; i < n; i++){
                string newStr = s.substr(0, i) + t[i] + s.substr(i + 1, n - i - 1);

                if(newStr == t){
                    return true;
                }
            }

            return false;
        } else if (n == m - 1){
            for(int i = 0; i < m; i++){
                string newStr = s.substr(0, i) + t[i] + s.substr(i, n - i);

                if(newStr == t){
                    return true;
                }
            }

            return false;
        } else if (n - 1 == m){
            for(int i = 0; i < n; i++){
                string newStr = s.substr(0, i) + s.substr(i + 1, n - i - 1);

                if(newStr == t){
                    return true;
                }
            }

            return false;
        } else {
            return false;
        }
    }
};

int main() {

    Solution sol;

    string source = "";
    string target = "";

    printf("isOneEditDistance: %d", sol.isOneEditDistance(source ,target));

    return 0;
}